lib/3scale/backend/storage_async/methods.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module Backend
+    module StorageAsync
+      module Methods
+        # Now we are going to define the methods to run redis commands
+        # following the interface of the redis-rb lib.
+        #
+        # These are the different cases:
+        #   1) Methods that can be called directly. For example SET:
+        #      call('SET', some_key)
+        #   2) Methods that need to be "boolified". These are methods for which
+        #      redis-rb returns a boolean, but redis just returns an integer.
+        #      For example, Redis returns 0 or 1 for the EXISTS command, but
+        #      redis-rb transforms that into a boolean.
+        #   3) There are a few methods that need to be treated differently and
+        #      do not fit in any of the previous categories. For example, SSCAN
+        #      which accepts a hash of options in redis-rb.
+        #
+        # All of this might be simplified a bit in the future using
+        # https://github.com/socketry/protocol-redis
```

Comment:
I don't understand how it would help but at least link opens :smile:

Reply:
So `protocol-redis` implements a ruby method for each redis command, which ends up calling `call [COMMAND], [ARGS]`. AFAIK Both `async-redis` and `redis-rb` implement the `:call` method so theoretically it would be possible to put `protocol-redis` in the middle and make apisonator use the `procol-redis` API rather than the `redis-rb` API. Which would allow us to remove this class entirely. However I didn't investigate enough to be sure it's possible.

Reply:
Thanks for the explanation @jlledom By looking at the README of that repo I also didn't understand... 
But now I understand.
Another link for reference (NOT saying that it needs to be added to the comment). https://www.rubydoc.info/gems/protocol-redis/0.2.0/Protocol/Redis/Methods/Hashes